test/functional/wallet_avoidreuse.py

```diff
@@ -288,5 +290,36 @@ def test_getbalances_used(self):
         assert_unspent(self.nodes[1], total_count=2, total_sum=6, reused_count=1, reused_sum=1)
         assert_balances(self.nodes[1], mine={"used": 1, "trusted": 5})
 
+    def test_large_destination_group(self):
+        '''
+        Test the case where [1] only has 11 outputs of 1 BTC in the same reused
+        address and tries to send a small payment of 0.5 BTC. The wallet
+        should use 10 outputs from the reused address as inputs and not a
+        single 1 BTC input, in order to join several outputs from the reused
+        address.
+        '''
```

Comment:
I think having a test where the code DOES use the discouraged <10 leftover group would be helpful to ensure that this is correctly chosen when necessary.

Reply:
Added it, thanks.